File: ncPersistencia/pEfdContabilidade.vb
Imports ncDados.nsEFD
Imports ncComum.nsAcessoBD
Imports ncComum.nsFuncoes
Imports ncComum.nsExcecao

Namespace nsEFD

    Public Class pEfdContabilidade

        Public Function Consultar() As dEfdContabilidade

            Dim retorno As dEfdContabilidade
            Dim acessoBanco As cAcessoBD
            Dim ds As DataSet
            Dim dt As DataTable
            Dim sqlSelect As String
            Dim sqlFrom As String

            Try

                acessoBanco = New cAcessoBD

                sqlSelect = " Select c.nomeContador, c.cpfContador, c.crcContador, c.cnpjEscritorio, c.logradouro, c.numero, " & _
                  " c.complemento, c.bairro, c.municipio, c.cep, c.dddTelefone, c.dddFax, c.email, c.contaAnaliticaContabil, m.estados_cid, " & _
                  " m.codigo_ibge "
                sqlFrom = " From EfdContabilidade c " & _
                  " left join municipios m " & _
                  "   on m.cid = c.municipio "

                ds = acessoBanco.ExecutarDS(sqlSelect & " " & sqlFrom)

                If Not ds Is Nothing Then
                    If ds.Tables.Count > 0 Then
                        dt = ds.Tables(0)

                        If dt.Rows.Count > 0 Then
                            retorno = New dEfdContabilidade

                            retorno.nomeContador = cFuncoes.RetornarTexto(dt.Rows(0).Item("nomeContador"))
                            retorno.cpf = cFuncoes.RetornarTexto(dt.Rows(0).Item("cpfContador"))
                            retorno.crc = cFuncoes.RetornarTexto(dt.Rows(0).Item("crcContador"))
                            retorno.cnpjEscritorio = cFuncoes.RetornarTexto(dt.Rows(0).Item("cnpjEscritorio"))
                            retorno.logradouro = cFuncoes.RetornarTexto(dt.Rows(0).Item("logradouro"))
                            retorno.numero = cFuncoes.RetornarTexto(dt.Rows(0).Item("numero"))
                            retorno.complemento = cFuncoes.RetornarTexto(dt.Rows(0).Item("complemento"))
                            retorno.bairro = cFuncoes.RetornarTexto(dt.Rows(0).Item("bairro"))
                            retorno.municipio = cFuncoes.RetornarTexto(dt.Rows(0).Item("municipio"))
                            retorno.municipioCodigoIbge = cFuncoes.RetornarTexto(dt.Rows(0).Item("codigo_ibge"))
                            retorno.cep = cFuncoes.RetornarTexto(dt.Rows(0).Item("cep"))
                            retorno.dddTelefone = cFuncoes.RetornarTexto(dt.Rows(0).Item("dddTelefone"))
                            retorno.dddFax = cFuncoes.RetornarTexto(dt.Rows(0).Item("dddFax"))
                            retorno.email = cFuncoes.RetornarTexto(dt.Rows(0).Item("email"))
                            retorno.contaAnaliticaContabil = cFuncoes.RetornarTexto(dt.Rows(0).Item("contaAnaliticaContabil"))
                            retorno.estados_cid = cFuncoes.RetornarInteiro(dt.Rows(0).Item("estados_cid"))
                        Else
                            retorno = Nothing
                        End If
                    Else
                        retorno = Nothing
                    End If
                Else
                    retorno = Nothing
                End If

            Catch ex As Exception

                retorno = Nothing
                Throw New ExcecaoNascomercio("Erro em Consultar EfdContabilidade [" & Me.ToString() & "] - " & ex.Message)

            End Try

            Consultar = retorno

        End Function

        Public Function Excluir() As Integer

            Dim retorno As Integer
            Dim acessoBanco As cAcessoBD
            Dim comandoSQL As String

            Try

                acessoBanco = New cAcessoBD

                comandoSQL = " DELETE FROM EfdContabilidade "

                retorno = acessoBanco.ExecutarINT(comandoSQL)

            Catch ex As Exception

                retorno = Nothing
                Throw New ExcecaoNascomercio("Erro em Excluir EfdContabilidade [" & Me.ToString() & "] - " & ex.Message)

            End Try

            Excluir = retorno

        End Function

        Public Function Incluir(ByVal dados As dEfdContabilidade) As Integer

            Dim retorno As Integer
            Dim acessoBanco As cAcessoBD
            Dim comandoSQL As String

            Try

                acessoBanco = New cAcessoBD

                comandoSQL = " INSERT INTO " & _
                    " EfdContabilidade ( nomeContador, cpfContador, crcContador, cnpjEscritorio, logradouro, " & _
                    " numero, complemento, bairro, municipio, cep, dddTelefone, dddFax, email, contaAnaliticaContabil " & _
                    " ) " & _
                    " VALUES (" & _
                    cFuncoes.PersistirTexto(dados.nomeContador) & "," & _
                    cFuncoes.PersistirTexto(dados.cpf) & "," & _
                    cFuncoes.PersistirTexto(dados.crc) & "," & _
                    cFuncoes.PersistirTexto(dados.cnpjEscritorio) & "," & _
                    cFuncoes.PersistirTexto(dados.logradouro) & "," & _
                    cFuncoes.PersistirTexto(dados.numero) & "," & _
                    cFuncoes.PersistirTexto(dados.complemento) & "," & _
                    cFuncoes.PersistirTexto(dados.bairro) & "," & _
                    cFuncoes.PersistirTexto(dados.municipio) & "," & _
                    cFuncoes.PersistirTexto(dados.cep) & "," & _
                    cFuncoes.PersistirTexto(dados.dddTelefone) & "," & _
                    cFuncoes.PersistirTexto(dados.dddFax) & "," & _
                    cFuncoes.PersistirTexto(dados.email) & "," & _
                    cFuncoes.PersistirTexto(dados.contaAnaliticaContabil) & ")"

                retorno = acessoBanco.ExecutarCID(comandoSQL)

            Catch ex As Exception

                retorno = Nothing
                Throw New ExcecaoNascomercio("Erro em Incluir EfdContabilidade [" & Me.ToString() & "] - " & ex.Message)

            End Try

            Incluir = retorno

        End Function

        Public Function Alterar(ByVal dados As dEfdContabilidade) As Integer

            Dim retorno As Integer
            Dim acessoBanco As cAcessoBD
            Dim comandoSQL As String

            Try

                acessoBanco = New cAcessoBD

                comandoSQL = " UPDATE EfdContabilidade SET " & _
                    " nomeContador = " & cFuncoes.PersistirTexto(dados.nomeContador) & "," & _
                    " cpfContador = " & cFuncoes.PersistirTexto(dados.cpf) & "," & _
                    " crcContador = " & cFuncoes.PersistirTexto(dados.crc) & "," & _
                    " cnpjEscritorio = " & cFuncoes.PersistirTexto(dados.cnpjEscritorio) & "," & _
                    " logradouro = " & cFuncoes.PersistirTexto(dados.logradouro) & "," & _
                    " numero = " & cFuncoes.PersistirTexto(dados.numero) & "," & _
                    " complemento = " & cFuncoes.PersistirTexto(dados.complemento) & "," & _
                    " bairro = " & cFuncoes.PersistirTexto(dados.bairro) & "," & _
                    " municipio = " & cFuncoes.PersistirTexto(dados.municipio) & "," & _
                    " cep = " & cFuncoes.PersistirTexto(dados.cep) & "," & _
                    " dddTelefone = " & cFuncoes.PersistirTexto(dados.dddTelefone) & "," & _
                    " dddFax = " & cFuncoes.PersistirTexto(dados.dddFax) & "," & _
                    " email = " & cFuncoes.PersistirTexto(dados.email) & "," & _
                    " contaAnaliticaContabil = " & cFuncoes.PersistirTexto(dados.contaAnaliticaContabil)

                retorno = acessoBanco.ExecutarINT(comandoSQL)

            Catch ex As Exception

                retorno = Nothing
                Throw New ExcecaoNascomercio("Erro em Alterar EfdContabilidade [" & Me.ToString() & "] - " & ex.Message)

            End Try

            Alterar = retorno

        End Function

    End Class

End Namespace

File: ncRepositorios/pEfdContabilidade.cs
using System;
using System.Data;
using ncNComum.nsAcessoBD;
using ncNComum.nsExcecao;
using static ncNComum.nsFuncoes.cFuncoes;
using nsEFD;

namespace ncPersistencia.nsEFD
{
    public class pEfdContabilidade
    {
        public dEfdContabilidade Consultar()
        {
            dEfdContabilidade retorno = null;
            try
            {
                var acessoBanco = new cAcessoBD();
                string sqlSelect = " Select c.nomeContador, c.cpfContador, c.crcContador, c.cnpjEscritorio, c.logradouro, c.numero, " +
                    " c.complemento, c.bairro, c.municipio, c.cep, c.dddTelefone, c.dddFax, c.email, c.contaAnaliticaContabil, m.estados_cid, " +
                    " m.codigo_ibge ";
                string sqlFrom = " From EfdContabilidade c " +
                    " left join municipios m " +
                    "   on m.cid = c.municipio ";
                var ds = acessoBanco.ExecutarDS(sqlSelect + " " + sqlFrom);
                if (ds != null && ds.Tables.Count > 0)
                {
                    DataTable dt = ds.Tables[0];
                    if (dt.Rows.Count > 0)
                    {
                        retorno = new dEfdContabilidade();
                        retorno.nomeContador = RetornarTexto(dt.Rows[0]["nomeContador"]);
                        retorno.cpf = RetornarTexto(dt.Rows[0]["cpfContador"]);
                        retorno.crc = RetornarTexto(dt.Rows[0]["crcContador"]);
                        retorno.cnpjEscritorio = RetornarTexto(dt.Rows[0]["cnpjEscritorio"]);
                        retorno.logradouro = RetornarTexto(dt.Rows[0]["logradouro"]);
                        retorno.numero = RetornarTexto(dt.Rows[0]["numero"]);
                        retorno.complemento = RetornarTexto(dt.Rows[0]["complemento"]);
                        retorno.bairro = RetornarTexto(dt.Rows[0]["bairro"]);
                        retorno.municipio = RetornarTexto(dt.Rows[0]["municipio"]);
                        retorno.municipioCodigoIbge = RetornarTexto(dt.Rows[0]["codigo_ibge"]);
                        retorno.cep = RetornarTexto(dt.Rows[0]["cep"]);
                        retorno.dddTelefone = RetornarTexto(dt.Rows[0]["dddTelefone"]);
                        retorno.dddFax = RetornarTexto(dt.Rows[0]["dddFax"]);
                        retorno.email = RetornarTexto(dt.Rows[0]["email"]);
                        retorno.contaAnaliticaContabil = RetornarTexto(dt.Rows[0]["contaAnaliticaContabil"]);
                        retorno.estados_cid = RetornarInteiro(dt.Rows[0]["estados_cid"]);
                    }
                }
            }
            catch (Exception ex)
            {
                retorno = null;
                throw new ExcecaoNascomercio("Erro em Consultar EfdContabilidade [" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }

        public int Excluir()
        {
            int retorno = 0;
            try
            {
                var acessoBanco = new cAcessoBD();
                string comandoSQL = " DELETE FROM EfdContabilidade ";
                retorno = acessoBanco.ExecutarINT(comandoSQL);
            }
            catch (Exception ex)
            {
                retorno = 0;
                throw new ExcecaoNascomercio("Erro em Excluir EfdContabilidade [" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }

        public int Incluir(dEfdContabilidade dados)
        {
            int retorno = 0;
            try
            {
                var acessoBanco = new cAcessoBD();
                string comandoSQL = " INSERT INTO " +
                    " EfdContabilidade ( nomeContador, cpfContador, crcContador, cnpjEscritorio, logradouro, " +
                    " numero, complemento, bairro, municipio, cep, dddTelefone, dddFax, email, contaAnaliticaContabil " +
                    " ) " +
                    " VALUES (" +
                    PersistirTexto(dados.nomeContador) + "," +
                    PersistirTexto(dados.cpf) + "," +
                    PersistirTexto(dados.crc) + "," +
                    PersistirTexto(dados.cnpjEscritorio) + "," +
                    PersistirTexto(dados.logradouro) + "," +
                    PersistirTexto(dados.numero) + "," +
                    PersistirTexto(dados.complemento) + "," +
                    PersistirTexto(dados.bairro) + "," +
                    PersistirTexto(dados.municipio) + "," +
                    PersistirTexto(dados.cep) + "," +
                    PersistirTexto(dados.dddTelefone) + "," +
                    PersistirTexto(dados.dddFax) + "," +
                    PersistirTexto(dados.email) + "," +
                    PersistirTexto(dados.contaAnaliticaContabil) + ")";
                retorno = acessoBanco.ExecutarCID(comandoSQL);
            }
            catch (Exception ex)
            {
                retorno = 0;
                throw new ExcecaoNascomercio("Erro em Incluir EfdContabilidade [" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }

        public int Alterar(dEfdContabilidade dados)
        {
            int retorno = 0;
            try
            {
                var acessoBanco = new cAcessoBD();
                string comandoSQL = " UPDATE EfdContabilidade SET " +
                    " nomeContador = " + PersistirTexto(dados.nomeContador) + "," +
                    " cpfContador = " + PersistirTexto(dados.cpf) + "," +
                    " crcContador = " + PersistirTexto(dados.crc) + "," +
                    " cnpjEscritorio = " + PersistirTexto(dados.cnpjEscritorio) + "," +
                    " logradouro = " + PersistirTexto(dados.logradouro) + "," +
                    " numero = " + PersistirTexto(dados.numero) + "," +
                    " complemento = " + PersistirTexto(dados.complemento) + "," +
                    " bairro = " + PersistirTexto(dados.bairro) + "," +
                    " municipio = " + PersistirTexto(dados.municipio) + "," +
                    " cep = " + PersistirTexto(dados.cep) + "," +
                    " dddTelefone = " + PersistirTexto(dados.dddTelefone) + "," +
                    " dddFax = " + PersistirTexto(dados.dddFax) + "," +
                    " email = " + PersistirTexto(dados.email) + "," +
                    " contaAnaliticaContabil = " + PersistirTexto(dados.contaAnaliticaContabil);
                retorno = acessoBanco.ExecutarINT(comandoSQL);
            }
            catch (Exception ex)
            {
                retorno = 0;
                throw new ExcecaoNascomercio("Erro em Alterar EfdContabilidade [" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }
    }
}
